### ISSUE TITLE
Health Check Topic growth

### ISSUE THREAD

**Issue by irairdon** [apache/pulsar#7559]
**Describe the bug**
If a node restart occurs the health check topic backlog will start to grow and appears to never stop growing

**To Reproduce**
1. Install pulsar 2.5.2 latest with Node restart fix
2. Start producer and consumer on any number of topics
3. restart a node in the kubernetes cluster
4. Observe backlog growth of one of the broker health check topics

**Expected behavior**
Health check is internal topics so I would expect pulsar to manage the backlog of internal topic without extra configuration

**Screenshots**
![clb_picture](https://user-images.githubusercontent.com/18467252/87698009-21680e00-c750-11ea-91bf-bbc028687522.png)

**Desktop (please complete the following information):**

**Additional context**
Here is the stats and stats-internal for the topic growing.

pulsar@itomdipulsar-bastion-7587f6cf5c-klwhx:/pulsar> bin/pulsar-admin topics stats persistent://pulsar/itomdipulsar/172.16.89.64:8080/healthcheck
{
  "msgRateIn" : 0.0,
  "msgThroughputIn" : 0.0,
  "msgRateOut" : 0.20000056635827046,
  "msgThroughputOut" : 20.200057202185317,
  "bytesInCounter" : 4045630,
  "msgInCounter" : 40148,
  "bytesOutCounter" : 748410,
  "msgOutCounter" : 7410,
  "averageMsgSize" : 0.0,
  "storageSize" : 751137,
  "backlogSize" : 748410,
  "publishers" : [ ],
  "subscriptions" : {
    "reader-f7da745e7f" : {
      "msgRateOut" : 0.20000056635827046,
      "msgThroughputOut" : 20.200057202185317,
      "bytesOutCounter" : 748410,
      "msgOutCounter" : 7410,
      "msgRateRedeliver" : 0.0,
      "msgBacklog" : 7410,
      "msgBacklogNoDelayed" : 7410,
      "blockedSubscriptionOnUnackedMsgs" : false,
      "msgDelayed" : 0,
      "unackedMessages" : 0,
      "type" : "Exclusive",
      "activeConsumerName" : "9d046",
      "msgRateExpired" : 0.0,
      "lastExpireTimestamp" : 0,
      "lastConsumedFlowTimestamp" : 1594914513257,
      "lastConsumedTimestamp" : 1594916563294,
      "lastAckedTimestamp" : 0,
      "consumers" : [ {
        "msgRateOut" : 0.20000056635827046,
        "msgThroughputOut" : 20.200057202185317,
        "bytesOutCounter" : 748410,
        "msgOutCounter" : 7410,
        "msgRateRedeliver" : 0.0,
        "consumerName" : "9d046",
        "availablePermits" : 590,
        "unackedMessages" : 0,
        "blockedConsumerOnUnackedMsgs" : false,
        "lastAckedTimestamp" : 0,
        "lastConsumedTimestamp" : 1594916563294,
        "metadata" : { },
        "clientVersion" : "2.5.2",
        "connectedSince" : "2020-07-16T06:05:20.258Z",
        "address" : "/172.16.89.64:38240"
      } ],
      "isReplicated" : false
    }
  },
  "replication" : { },
  "deduplicationStatus" : "Disabled"
}
pulsar@itomdipulsar-bastion-7587f6cf5c-klwhx:/pulsar> bin/pulsar-admin topics stats-internal persistent://pulsar/itomdipulsar/172.16.89.64:8080/healthcheck
{
  "entriesAddedCounter" : 36163,
  "numberOfEntries" : 7446,
  "totalSize" : 752046,
  "currentLedgerEntries" : 9,
  "currentLedgerSize" : 909,
  "lastLedgerCreatedTimestamp" : "2020-07-16T16:22:48.297Z",
  "lastLedgerCreationFailureTimestamp" : "2020-07-16T06:05:20.045Z",
  "waitingCursorsCount" : 2,
  "pendingAddEntriesCount" : 0,
  "lastConfirmedEntry" : "73889:8",
  "state" : "LedgerOpened",
  "ledgers" : [ {
    "ledgerId" : 59169,
    "entries" : 27,
    "size" : 2727,
    "offloaded" : false
  }, {
    "ledgerId" : 59211,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59293,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59346,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59395,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59447,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59500,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59549,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59602,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59652,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59700,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59756,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59809,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59861,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59922,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 59968,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60022,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60068,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60123,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60174,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60230,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60278,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60328,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60379,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60426,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60479,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60530,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60585,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60635,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60683,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60732,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60780,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60834,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60889,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60938,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 60986,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61043,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61088,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61142,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61189,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61244,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61310,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61361,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61414,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61457,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61503,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61555,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61604,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61659,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61714,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61766,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61815,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61871,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61923,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 61969,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62024,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62079,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62132,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62181,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62231,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62279,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62331,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62387,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62437,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62486,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62541,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62591,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62634,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62689,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62742,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62794,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62851,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62899,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 62953,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63003,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63054,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63106,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63153,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63206,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63261,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63315,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63367,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63414,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63459,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63505,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63563,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63614,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63667,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63717,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63778,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63825,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63875,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63930,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 63974,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64027,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64077,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64129,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64191,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64243,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64286,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64338,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64391,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64444,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64497,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64550,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64599,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64646,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64691,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64742,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64800,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64855,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64906,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 64954,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65003,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65050,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65108,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65163,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65214,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65263,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65313,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65359,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65405,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65463,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65509,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65569,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65623,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65672,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65716,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65770,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65822,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65870,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65921,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 65973,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66027,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66083,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66134,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66188,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66229,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66279,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66333,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66385,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66436,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66489,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66540,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66586,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66638,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66690,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66739,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66790,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66845,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66893,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 66940,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67006,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67056,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67110,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67163,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67218,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67267,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67318,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67357,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67414,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67471,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67515,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67566,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67617,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67669,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67712,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67763,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67819,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67873,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67926,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 67975,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68022,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68073,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68123,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68179,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68232,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68284,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68333,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68384,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68435,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68483,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68541,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68595,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68647,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68697,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68748,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68794,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68846,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68897,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68946,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 68998,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69050,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69104,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69160,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69202,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69256,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69303,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69356,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69408,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69465,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69521,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69566,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69610,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69657,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69710,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69764,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69823,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69878,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69929,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 69979,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70026,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70082,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70134,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70181,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70233,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70287,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70334,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70386,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70435,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70485,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70536,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70589,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70641,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70698,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70750,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70803,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70848,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70897,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 70952,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71003,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71057,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71104,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71157,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71205,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71282,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71333,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71381,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71435,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71490,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71537,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71588,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71635,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71686,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71737,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71791,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71841,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71891,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71942,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 71991,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72041,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72090,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72139,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72192,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72244,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72296,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72344,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72390,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72444,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72496,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72554,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72604,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72654,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72715,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72758,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72814,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72864,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72919,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 72971,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73021,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73073,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73117,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73172,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73224,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73273,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73325,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73378,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73424,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73479,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73523,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73578,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73636,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73686,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73740,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73792,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73835,
    "entries" : 26,
    "size" : 2626,
    "offloaded" : false
  }, {
    "ledgerId" : 73889,
    "entries" : 0,
    "size" : 0,
    "offloaded" : false
  } ],
  "cursors" : {
    "reader-f7da745e7f" : {
      "markDeletePosition" : "59169:26",
      "readPosition" : "73889:9",
      "waitingReadOp" : true,
      "pendingReadOps" : 0,
      "messagesConsumedCounter" : 28744,
      "cursorLedger" : -1,
      "cursorLedgerLastEntry" : -1,
      "individuallyDeletedMessages" : "[]",
      "lastLedgerSwitchTimestamp" : "2020-07-16T06:05:20.258Z",
      "state" : "Uninitialized",
      "numberOfEntriesSinceFirstNotAckedMessage" : 7420,
      "totalNonContiguousDeletedMessagesRange" : 0,
      "properties" : { }
    }
  }
}

**Comment by sijie** [comment]
@irairdon Can you also give us the broker log and heap dump if possible?

**Comment by codelipenghui** [comment]
@irairdon Could you please help check if there are logs contains "Error closing reader for healthcheck" in the broker log?

**Comment by irairdon** [comment]
So I just looked at broker logs and I do not see the error message you have ask about. I am attaching all broker logs here.
[itomdipulsar-broker-5c85589ccc-kvzgr.log](https://github.com/apache/pulsar/files/4938730/itomdipulsar-broker-5c85589ccc-kvzgr.log)
[itomdipulsar-broker-5c85589ccc-9bbzh.log](https://github.com/apache/pulsar/files/4938731/itomdipulsar-broker-5c85589ccc-9bbzh.log)
[itomdipulsar-broker-5c85589ccc-578vj.log](https://github.com/apache/pulsar/files/4938732/itomdipulsar-broker-5c85589ccc-578vj.log)


I am working on grabbing the heap as well but that may need to be uploaded in slack as I am not sure github UI can handle that large of a file.

**Comment by codelipenghui** [comment]
Thanks @irairdon

**Comment by codelipenghui** [comment]
@irairdon I have checked the heap dump, the problematic reader is `reader-08cbbdc976`. Looks the reader is closed at the client side but not clean up at the broker side. But I can't find any logs related to this reader, the last time of the broker log is `14:54:13.528` and `reader-08cbbdc976` is created later than `15:00:00`. I just inferred from the subscription timeout. Could you please send me a completed log of broker `578vj`? I suspect a race condition now but need more logs to confirm.

**Comment by irairdon** [comment]
@codelipenghui after merging PR 7589 to our Github Enterprise repo and building locally it looks like that code change has resolved the issue we had seen.  I will leave to you to close the issue here once it is merged into master, 2.6, 2.5 and any other place this should be merged to.
